app/web/features/search/SearchPage.tsx

```diff
@@ -52,193 +55,195 @@ const useStyles = makeStyles((theme) => ({
-    }
-  }, [selectedResult, areClustersLoaded, previousResult, flyToUser]);
-
-  const handleMapUserClick = useCallback(
-    (
-      ev: maplibregl.MapMouseEvent & {
-        features?: maplibregl.MapboxGeoJSONFeature[] | undefined;
-      } & EventData
-    ) => {
-      ev.preventDefault();
-      const props = ev.features?.[0].properties;
-      const geom = ev.features?.[0].geometry as Point;
-      if (!props || !geom) return;
-      const username = props.username;
-      const userId = props.id;
-      const [lng, lat] = geom.coordinates;
-      setSelectedResult({ username, userId, lng, lat });
-    },
-    []
-  );
-
-  //detect when map data has been initially loaded
-  const handleMapSourceData = useCallback(() => {
-    if (
-      map.current &&
-      map.current.getSource("clustered-users") &&
-      map.current.isSourceLoaded("clustered-users")
-    ) {
-      setAreClustersLoaded(true);
-
-      // unbind the event
-      map.current.off("sourcedata", handleMapSourceData);
-    }
-  }, []);
-
-  useEffect(() => {
-    if (!map.current) return;
-    const handleMapClickAway = (e: EventData) => {
-      //defaultPrevented is true when a map feature has been clicked
-      if (!e.defaultPrevented) {
-        setSelectedResult(undefined);
+        return service.search.userSearch(
+          {
+            query: queryName,
+            bbox: locationResult.bbox,
+            lastActive:
+              lastActiveComparation === 0 ? undefined : lastActiveFilter,
+            hostingStatusOptions:
+              hostingStatusFilterComparation === 0
+                ? undefined
+                : [hostingStatusFilter],
+            numGuests: numberOfGuestFilter,
+            completeProfile:
+              completeProfileFilter === false
+                ? undefined
+                : completeProfileFilter,
+          },
+          pageParam
+        );
+      },
+      {
+        getNextPageParam: (lastPage) =>
+          lastPage.nextPageToken ? lastPage.nextPageToken : undefined,
       }
-    };
-
-    //bind event handlers for map events (order matters!)
-    map.current.on(
-      "click",
-      layers.unclusteredPointLayer.id,
-      handleMapUserClick
     );
-    map.current.on("click", handleMapClickAway);
 
-    map.current.on("sourcedata", handleMapSourceData);
-
-    return () => {
-      if (!map.current) return;
+  // Relocate map everytime boundingbox changes
+  useEffect(() => {
+    map.current?.fitBounds(locationResult.bbox, {
+      maxZoom: selectedUserZoom,
+    });
+  }, [locationResult.bbox]);
 
-      //unbind event handlers for map events
-      map.current.off("sourcedata", handleMapSourceData);
-      map.current.off("click", handleMapClickAway);
-      map.current.off(
-        "click",
-        layers.unclusteredPointLayer.id,
-        handleMapUserClick
+  // Initial bounding box
+  useEffect(() => {
+    if (isLoadingUser === false && userData && bbox.join() === "0,0,0,0") {
```

Comment:
`bbox.join() === "0,0,0,0"` could potentially be a lot of logic if the array is large. Maybe make this bit a variable and reference it here instead?

Reply:
hmm the thing it's that bbox its a 4 lenght array that we receive from the openstreetmap API, is always 4 length array since it represents the top left/right corner & bottom left/right corner, basically what the user sees in it's screen (so it's always 4 values)

so I have the confidence that the array it's always 4 digits long (so `O(1)` in that case)